cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -179,6 +188,35 @@ func (a *StaticAutoscaler) cleanUpIfRequired() {
 	a.initialized = true
 }
 
+func (a *StaticAutoscaler) propagateClusterSnapshot(nodes []*apiv1.Node, scheduledPods []*apiv1.Pod) errors.AutoscalerError {
+	var err error
+	err = a.ClusterSnapshot.Clear()
+	if err != nil {
+		klog.Errorf("Failed to clear cluster snapshot: %v", err)
+		return errors.ToAutoscalerError(errors.InternalError, err)
```

Comment:
That seems like it would be unrecoverable if it ever happened (which seems impossible with current implementation of snapshot, so we can probably leave it as is for now). If that actually is a possibility I would consider adding a CriticalError that triggers restart or something along those lines.

Reply:
Error from `Clear` will be dropped (there is a followup for that). We might also drop `Clear` at all and create new `ClusterSnapshot` when needd.